core/kernel/pseudo_ta.c

```diff
@@ -294,6 +294,12 @@ TEE_Result tee_ta_init_pseudo_ta_session(const TEE_UUID *uuid,
 	struct tee_ta_ctx *ctx;
 	const struct pseudo_ta_head *ta;
 
+	/*
+	 * Caller is expected to hold tee_ta_mutex for safe changes
+	 * in @s and registering of the context in tee_ctxes list.
+	 */
+	assert(!mutex_trylock(&tee_ta_mutex));
```

Comment:
This is not very nice even if it gives a somewhat usable behaviour. I don't want to risk getting this pattern copied.
Please remove this assert if `mutex_is_locked()` isn't acceptable.

Reply:
Do you mean you'd rather have this line replaced by `assert(mutex_is_locked())`?

As `mutex_is_locked()` is likely not to be very bullet-proof but for weak debug purpose, i wonder if a debug helper `assert_mutex_is_locked()` would be more suitable.

Reply:
> Do you mean you'd rather have this line replaced by `assert(mutex_is_locked())`?

Yes, either that or removed.

> 
> As `mutex_is_locked()` is likely not to be very bullet-proof but for weak debug purpose, i wonder if a debug helper `assert_mutex_is_locked()` would be more suitable.

Nor is `mutex_trylock()`. If the mutex is already held by another thread it will do no better than `mutex_is_locked(). If the mutex is unlocked it will have a side effect before the assert triggers. It may also lead to confusion about whether there's a missing `mutex_unlock()` or if it's intended that `mutex_trylock()` only should be called when debug is enabled.

I only want a reminder that the mutex must be held here. I thought `assert(mutex_is_locked())` would be a nice and easy complement to that but it seems it was neither.

Reply:
series updated (and rebased for CI support)